### PR TITLE
Adds caks

### PR DESCRIPTION
**Pull request by ghost**
:cl: Xhuis
rscadd: You can now create living cake/cat hybrids through a slightly expensive crafting recipe. These "caks" are remarkably resilient, quickly regenerating any brute damage dealt, and can be attacked on harm intent to take a bite to be fed by a small amount. They serve as mobile sources of food and make great pets.
rscadd: If a brain occupied by a player is used in the crafting recipe for caks, that player then takes control of the cak!
/ :cl:

Adds caks - cat/cake hybrids, as requested by Coderbus.
- 50 max health
  - Regenerates brute damage at a rate of 8/tick
- Takes 10 brute damage when hit with a harm intent attack
  - Harm intent attacks take a bite from the cak, adding nutriment and vitamin to the attacker
- Created by crafting a brain, heart, birthday cake, three slabs of meat, 30u of blood, 5u of sprinkles, and 1u of teslium
  - If the brain is occupied by a player, the player is placed in control of the cak
- Butchering drops some (but not all) of the components used in creating the cak
  - Brains dropped in this manner lose sentience

meow

**_Token:_ Remaining fix from #18387**

### CODE REVIEW

**Comment by KorPhaeron** [comment]
You can't use stuff from before may 15th

**Comment by Xhuis** [comment]
All right, thanks. I'll whip up a fix for use as a token.

File: code/modules/mob/living/simple_animal/friendly/cat.dm [hunk 1]
```diff
@@ -212,3 +212,32 @@
 		else
 			if(M && stat != DEAD)
 				emote("me", 1, "hisses!")
+
+/mob/living/simple_animal/pet/cat/cak //I told you I'd do it, Remie
```
Comment (RemieRichards) [inline code/modules/mob/living/simple_animal/friendly/cat.dm]: that you did, that you did.

**Comment by Razharas** [comment]
Wonderful

**Comment by WJohn** [comment]
why did you make this

i didn't know i wanted it

**Comment by RemieRichards** [comment]
he accidentally wrote "cak", and then rolled with it.

File: code/modules/mob/living/simple_animal/friendly/cat.dm [hunk 2]
```diff
@@ -212,3 +212,32 @@
...
+	/obj/item/weapon/reagent_containers/food/snacks/meat = 2)
+	response_harm = "takes a bite out of"
+	attacked_sound = 'sound/items/eatfood.ogg'
+
+/mob/living/simple_animal/pet/cat/cak/Life()
+	..()
+	if(health < maxHealth)
+		adjustBruteLoss(-2) //Fast life regen
+	for(var/obj/item/weapon/reagent_containers/food/snacks/donut/D in range(1, src)) //Frosts nearby donuts!
+		if(D.name != "frosted donut")
```
Comment (phil235) [inline code/modules/mob/living/simple_animal/friendly/cat.dm]: use hand labeller or pen for infinite reagents.

Comment (Xhuis) [inline code/modules/mob/living/simple_animal/friendly/cat.dm]: feature

**Comment by Zilenan91** [comment]
t units of sprinkles?

**Comment by Xhuis** [comment]
5, was a typo

**Comment by SidonCC** [comment]
I love caks

**Comment by PKPenguin321** [comment]
i'm not sure it's possible to get sprinkles and a birthday cake, as there is only one cake hat and getting sprinkles requires you to destroy the cake

**Comment by Xhuis** [comment]
Sprinkles can come from frosted donuts.

**Comment by Core0verload** [comment]
Or botany, because cell charge shrooms come with some teslium inside.

**Comment by Xhuis** [comment]
Oh yeah, that's true.

**Comment by Core0verload** [comment]
Maybe make it require Strange Reagent? It's used for reviving and creating life.

Also, would using player's brain in recipe transfer mind to cak? Would be fun to debrain jerks and bring them to life as mad science cat/cake hybrids that everyone is trying to eat.

**Comment by Xhuis** [comment]
I _really_ wanted to do that second part but I unfortunately didn't really know how. As for strange reagent, I wanted it to be _Frankenstein_-esque - an amalgamation of body parts brought to life by a bunch of electricity/lightning.

**Comment by Core0verload** [comment]
See cyborg or AI code for some how-to on mind transfer. See advances crafting recipes such as pin removal for how to pass ingredients down to crafted object.

**Comment by Xhuis** [comment]
I guess I'm not seeing it. Although I know how to do mind transfer, crafting code befuddles me.

**Comment by Core0verload** [comment]
Crafting code is a mess.

Any items that are both in `reqs` and `parts` of recipe are put in `result`'s `CheckParts(list/parts_list)` proc. Intercept a brain there and transfer mind, then qdel brain. Don't forget to `..()`.

**Comment by Xhuis** [comment]
Thanks, I'll see what I can do.

**Comment by Xhuis** [comment]
_New in this commit:_
* If the brain used for a cak is occupied by a player, that player is placed in control of the cak and can change their name. They're advised to spread joy and cake to everyone on the station.
* Caks now have a death sound and message.
* Caks now have 1000 health and regenerate at a rate of 150/s. Taking bites out of caks deals 200 brute damage, thus making them resilient to pretty much anything else. _(Note: If this is over-the-top, I'll change it.)_
* Caks properly drop 2 slabs of meat upon butchering.

**Comment by PKPenguin321** [comment]
that hp/hp regen probably means caks are extremely good at recovering bodies from fires/space, i don't know if we want that

**Comment by Xhuis** [comment]
\>implying anyone gets bodies from space anyway

**Comment by Xhuis** [comment]
_New in this commit:_
* Nerfed cak health to a reasonable amount (50 max, 8/s regen and 10/bite)

**Comment by optimumtact** [comment]
:cake: :cat2:

**Pull request merged**